Initial Classes

mecha opened the pull request.

mecha committed:
Installed required dependencies

mecha committed:
Aliased `develop` branch to `0.1.x-dev`

mecha committed:
Added project test configuration

- PHPStorm testing config
- Composer autoloader config for test dirs

mecha committed:
Imported unit tests from old package

mecha committed:
Imported sources from old package

XedinUnknown reviewed (changes requested):
Looks like at least one of the traits addresses more than one concern.

mecha committed:
Refactored `EscapeSqlReferencesCapableTrait` ...

- Renamed primary method
- Unified single and array escaping logic

mecha committed:
Fixed old uses of value escaping abstract method

mecha committed:
Added `NormalizeSqlValueCapableTrait`

mecha committed:
Fixed uses of old sanitization method with new ...

... value normalization.

mecha committed:
Fixed test name - missing `Test` suffix

mecha committed:
Added `BuildSqlRecordValuesCapableTrait`

mecha committed:
Fixed `BuildInsertSqlCapableTrait` to use new ...

... `_buildSqlRecordValues()` abstract method.

mecha committed:
Added `BuildSqlUpdateSetCapableTrait`

mecha committed:
Fixed `BuildUpdateSqlCapableTrait` to use new ...

... `_buildSqlUpdateSet()` abstract method.

mecha committed:
Fixed code standards

mecha committed:
Fixed SQL value normalization ...

- Added optional column param
- Numeric strings are left unquoted
- Invalid values cause an `OutOfRangeException` to be thrown

mecha committed:
Can now escape entity and field aware instances

mecha committed:
Extracted single reference escaping into a trait

mecha committed:
Refactored old SQL reference escaping trait ...

Now only accepts and escapes lists, using an abstracted single-reference escaping method.

mecha committed:
Updated dependencies

mecha committed:
Added `BuildSqlOrderByCapableTrait`

mecha committed:
Added `BuildSqlLimitCapableTrait`

mecha committed:
Added `BuildSqlOffsetCapableTrait`

mecha committed:
Renamed argument for clarity purposes

Because "orders" sounds like an e-commerce thing ...

mecha committed:
Reordered params for escaping references and ...

... renamed params to "reference" and "prefix".

mecha committed:
Moved incorrectly located test file

mecha committed:
Installed more required dependencies ...

- `dhii/data-container-interface` for exception interfaces
- `dhii/exception-interface` for exception interfaces

mecha committed:
Fields in ordering are now changed into columns

mecha committed:
Added ordering, limit and offset to DELETE

mecha committed:
Added `GetSqlColumnNameCapableTrait`

mecha committed:
Fixed missing method mock in test

mecha committed:
Fixed old reference escaping abstract methods

mecha committed:
Fixed code standards

mecha committed:
Added more tests for `BuildDeleteSqlCapableTrait`

mecha committed:
Improved building in `BuildDeleteSqlCapableTrait`

mecha committed:
Added ordering and limiting in ...

... `BuildSelectSqlCapableTrait` and improved query string building.

mecha committed:
Added ordering and limiting in ...

... `BuildUpdateSqlCapableTrait` and improved query string building.

mecha committed:
Fixed use of abstract `_escapeSqlReference()`

mecha committed:
Fixed incompatible abstract signature

mecha committed:
Added `BuildSqlFromCapableTrait`

mecha committed:
Table aliasing is optional

mecha committed:
Updated table list building for SELECT SQL

mecha committed:
Updated SQL list awareness trait

mecha committed:
Fixed merging of aliases for expression render

mecha committed:
Added `BuildSqlColumnListCapableTrait`

mecha committed:
Select SQL builder uses new column list building

mecha committed:
Swapped mapping for SQL FROM builder to match ...

... new column builder. Aliases are unique, and a table name may be used twice with different names.

mecha committed:
Updated doc and signature for `_buildSelectSql()`

mecha committed:
Capitalized "as" keywords to "AS"

mecha committed:
Fixed method incompatibility

mecha committed:
Field-column map setter normalizes, not validates

mecha committed:
Fixed SQL field name from map resolution

XedinUnknown committed:
Corrected readme

XedinUnknown committed:
Deps updated

XedinUnknown committed:
Now also building on PHP 7.2

XedinUnknown committed:
Added changelog entry

XedinUnknown committed:
Added unique prefixes to random values in some tests

mecha committed:
Added unique prefixes to random values in more tests
